commit b8d966efd9a46a9a35beac50cbff6e30565125ef
Author: NeilBrown <neilb@suse.de>
Date:   Wed Jul 1 11:14:04 2009 +1000

    md: avoid dereferencing NULL pointer when accessing suspend_* sysfs attributes.
    
    If we try to modify one of the md/ sysfs files
      suspend_lo or suspend_hi
    when the array is not active, we dereference a NULL.
    Protect against that.
    
    Cc: stable@kernel.org
    Signed-off-by: NeilBrown <neilb@suse.de>

Adjusted to apply to Debian's 2.6.18 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.18.orig/drivers/md/md.c linux-source-2.6.18/drivers/md/md.c
--- linux-source-2.6.18.orig/drivers/md/md.c	2009-08-15 12:05:07.000000000 -0600
+++ linux-source-2.6.18/drivers/md/md.c	2009-08-20 15:02:27.000000000 -0600
@@ -2772,7 +2772,8 @@ suspend_lo_store(mddev_t *mddev, const c
 	char *e;
 	unsigned long long new = simple_strtoull(buf, &e, 10);
 
-	if (mddev->pers->quiesce == NULL)
+	if (mddev->pers == NULL || 
+	    mddev->pers->quiesce == NULL)
 		return -EINVAL;
 	if (buf == e || (*e && *e != '\n'))
 		return -EINVAL;
@@ -2800,7 +2801,8 @@ suspend_hi_store(mddev_t *mddev, const c
 	char *e;
 	unsigned long long new = simple_strtoull(buf, &e, 10);
 
-	if (mddev->pers->quiesce == NULL)
+	if (mddev->pers == NULL ||
+	    mddev->pers->quiesce == NULL)
 		return -EINVAL;
 	if (buf == e || (*e && *e != '\n'))
 		return -EINVAL;
